File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/Go/s378126458_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)
	scanner.Scan()
	N, _ := strconv.Atoi(scanner.Text())
	scanner.Scan()
	Q, _ := strconv.Atoi(scanner.Text())
	scanner.Scan()
	s := scanner.Text()

	pre := make([]int, N+1)
	for i := 1; i < N; i++ {
		if s[i-1] == 'A' && s[i] == 'C' {
			pre[i] = 1
		}
	}
	// Build prefix sum
	for i := 1; i < N; i++ {
		pre[i] += pre[i-1]
	}
	for Q > 0 {
		Q--
		scanner.Scan()
		l, _ := strconv.Atoi(scanner.Text())
		scanner.Scan()
		r, _ := strconv.Atoi(scanner.Text())
		r--
		fmt.Println(pre[r] - pre[l-1])
	}
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/C++/s378126458_commented.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    int N, Q;
    cin >> N >> Q;
    string s;
    cin >> s;
    vector<int> pre(N + 1, 0);
    
    // Building the k array and prefix sum array in one loop
    for (int i = 1; i < N; i++) {
        pre[i] = (s[i - 1] == 'A' && s[i] == 'C') ? 1 : 0;
        pre[i] += pre[i - 1];
    }
    
    // Querying
    while (Q-- > 0) {
        int l, r;
        cin >> l >> r;
        cout << pre[r - 1] - pre[l - 1] << endl;
    }
    
    return 0;
}
// End of Code
